### PR TITLE
Validate input for hcall assign ptdev

### PR DESCRIPTION
**Pull request by yonghuah**

### CODE REVIEW

File: hypervisor/common/hypercall.c [hunk 1]
```diff
@@ -838,7 +838,7 @@ int32_t hcall_assign_ptdev(struct acrn_vm *vm, uint16_t vmid, uint64_t param)
 		if (vdev == NULL) {
 			pr_fatal("%s %x:%x.%x not found\n", __func__, bdf.bits.b,  bdf.bits.d,  bdf.bits.f);
 			ret = -EPERM;
-		} else {
+		} else if (vdev->pdev != NULL) {
```
Comment (dongyaozu) [inline hypervisor/common/hypercall.c]: Then we need another else {}, for MISRA-C compliance.

Comment (yonghuah) [reply hypervisor/common/hypercall.c]: @dongyaozu  updated with combining the condition check for "vdev" and “vdev", and adding reviewers , thanks.

**Comment by wenlingz** [comment]
OK to verify

**Comment by wenlingz** [comment]
OK to verify

**Comment by wenlingz** [comment]
@HuiHuangShi please help add MISRA-C label, thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**